packages/python-packages/api-stub-generator/apistub/_apiview.py

```diff
@@ -146,8 +148,12 @@ def _add_type_token(self, type_name, line_id = None):
         if navigate_to_id:
             token.NavigateToId = navigate_to_id
         elif type_name.startswith("~") and line_id:
-            # if navigation ID is missing for internal type, add diagnostic error
-            self.add_diagnostic(SOURCE_LINK_NOT_AVAILABLE.format(token.Value), line_id)            
+            # Check if type name is importable. If type name is incorrect in docstring then it wont be importable
+            # If type name is importable then it's a valid type name. Source link wont be available if type is from 
+            # different package
+            if not is_valid_type_name(type_full_name):
+                # Navigation ID is missing for internal type, add diagnostic error
+                self.add_diagnostic(SOURCE_LINK_NOT_AVAILABLE.format(token.Value), line_id)            
```

Comment:
Note for future: if we start validating the types in type hints against the types in docstrings, the type in the type hint will already by imported in the current context (i.e. it wont be fully qualified)

Reply:
Reason for adding this check is to filter out any any types in different packages. For e.g. docstring mention a type in azure-core package as type for a param in form recognizer package. ( I think it's pipeline related). But this type definition is out side the scope of form recognizer package. Added this validation to ensure qualified name in docstring is actually importable to find any incorrect link to type definition.